### ISSUE TITLE
Parameter order: Nested parameters are always behind non-nested parameters

### ISSUE THREAD

**Issue by sleiner** [robbert-vdh/nih-plug#43]
## How to reproduce

Have a plugin where the parameter struct contains "non-nested" parameters aft`r nested ones:

```rust
#[derive(Params)]
struct PluginParams {
    #[id = "one"]
    pub one: FloatParam,

    #[nested(group = "inner")]
    pub two_and_three: InnerParams,

    #[id = "four"]
    pub four: FloatParam,
}

#[derive(Params)]
struct InnerParams {
    #[id = "two"]
    pub two: FloatParam,

    #[id = "three"]
    pub three: FloatParam,
}
```

See also: [minimal example plugin](https://github.com/sleiner/nih-param-order-example)

## Expected behavior

When opening the plugin in the host, the parameters are displayed in this order:

1. One
2. Two
3. Three
4. Four

## Actual Behavior

The nested parameters from `InnerParams` are displayed _after_ all non-nested parameters:

1. One
2. Four
3. Two
4. Three

See, for example, Bitwig:
<img width="214" alt="Screenshot_2022-11-07_22-18-03" src="https://user-images.githubusercontent.com/6379313/200418972-57e58a70-b21c-4ec8-944b-30645d90a3b2.png">

**Comment by sleiner** [comment]
> this is the same order the parameters will appear in when the DAW uses a tree view (which in DAWs usually seem to display parameters above other subgroups).

I see, that makes sense... This means, that (aside from the macro implementation becoming more complicated :/) _non-grouped_ nested parameters (unlike the ones from the example above) would not have to be moved to the bottom, right?

**Comment by robbert-vdh** [comment]
They would not, no. This is just how the macro is implemented now. The parameter mapping vector is constructed using the `vec![]` macro for all fields at once, and then the parameter triples for the child structs are added to that vector. If you want to preserve arbitrary order you'd need to introduce another sum type and interleave the code gen.

**Comment by sleiner** [comment]
Closing this since the issue was addressed in #44 (and the subsequent patches)
